### PR TITLE
Small Rest API maintenance

### PR DESCRIPTION
**Pull request by Defi-Moses**
This PR cleans up some of the rest API config to match that replicated throughout sanguine (the usage of bridgeMap and bridgeable files). This also upgrades the existing API to the current version of the SDK and uses the new method of allBridgeQuotes, adding additional return information. 

This repo doesnt auto-deploy to heroku, but the changes are pushed, tested and live there. The changes are also backwards compatible. 

## Summary by CodeRabbit

- **New Features**
	- Improved token information retrieval logic in the REST API.
	- Enhanced swap and bridge quote endpoints with updated token information logic.
	- Updated swap and bridge transaction info endpoints for better error handling and efficiency.
- **Refactor**
	- Optimized token lookup by using a new `Token` model with extended properties.
	- Streamlined API routes for better performance and error handling.
	- Refactored internal logic for handling blockchain data, utilizing a new mapping system.
- **Documentation**
	- Introduced new configuration files for blockchain and token information, enhancing the API's extendibility and maintainability.

### CODE REVIEW

File: packages/rest-api/src/app.ts [hunk 1]
```diff
@@ -93,324 +118,400 @@ app.get('/', ({ res }) => {
...
         `
-      <h1>Invalid Request</h1>
-      <code>${err}</code>
-      <hr/>
-      <b>Ensure that your request matches the following format: /swapTxInfo?chain=1&fromToken=USDC&toToken=DAI&amount=100</b>
-      <h2>Available Tokens (symbols to use)</h2>
-      ${tokenHtml}`
+        <h1>Invalid Params</h1>
+        <hr/>
+        <b>Ensure that your request matches the following format: /bridge?fromChain=1&toChain=42161&fromToken=USDC&toToken=USDC&amount=1000000
```
Comment (ChiTimesChi) [inline packages/rest-api/src/app.ts:334]: Should be `swapTxInfo` here?

File: packages/rest-api/src/app.ts [hunk 2]
```diff
@@ -93,324 +118,400 @@ app.get('/', ({ res }) => {
...
+      !fromTokenAddress ||
+      !toTokenAddress ||
+      !fromTokenDecimals ||
+      !toTokenDecimals
+    ) {
+      res.send(
+        `
+        <h1>Invalid Params</h1>
+        <hr/>
+        <b>Ensure that your request matches the following format: /swap?chain=1&fromToken=USDC&toToken=DAI&amount=100</b>
```
Comment (ChiTimesChi) [inline packages/rest-api/src/app.ts:357]: `swapTxInfo`

File: packages/rest-api/src/app.ts [hunk 3]
```diff
@@ -93,324 +118,400 @@ app.get('/', ({ res }) => {
...
+
+    if (!fromTokenInfo || !toTokenInfo) {
       res.send(
         `
-        <h1>Invalid Request</h1>
-        <code>${err}</code>
+        <h1>Invalid Params</h1>
         <hr/>
-        <b>Ensure that your request matches the following format: /bridgeTxInfo?fromChain=1&toChain=42161&fromToken=USDC&toToken=USDC&amount=1000000&destAddress=0xcc78d2f004c9de9694ff6a9bbdee4793d30f3842</b>
+        <b>Ensure that your request matches the following format: /bridge?fromChain=1&toChain=42161&fromToken=USDC&toToken=USDC&amount=1000000
```
Comment (ChiTimesChi) [inline packages/rest-api/src/app.ts:430]: `bridgeTxInfo`

File: packages/rest-api/src/config/Chains.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  HARMONY: { id: 1666600000 },
+  MOONBEAM: { id: 1284 },
+  CRONOS: { id: 25 },
+  METIS: { id: 1088 },
+  DOGE: { id: 2000 },
+  CANTO: { id: 7700 },
+  KLAYTN: { id: 8217 },
+  DFK: { id: 53935 },
+  BASE: { id: 8453 },
+  AURORA: { id: 1313161554 },
```
Comment (ChiTimesChi) [inline packages/rest-api/src/config/Chains.ts:24]: Small nit: this is better maintained when sorted (either by id or chain alias)

File: packages/rest-api/src/config/bridgeMap.ts
Comment (ChiTimesChi) [inline packages/rest-api/src/config/bridgeMap.ts:1]: What is the reason for not using `synapse-constants` to pull this one and `bridgeable.ts`?

**Review by ChiTimesChi — APPROVED** [review]
LGTM, though outlined some inconsistencies in error messages.

File: packages/rest-api/src/config/bridgeMap.ts
Comment (Defi-Moses) [reply packages/rest-api/src/config/bridgeMap.ts:1]: Keep running into issues using the package within sanguine -- I updated the live version to use the package. This also fixes the chain ID issue above. Id love to hook this up to the actual deployed code but blocked there on perms and a workflow isnt that urgent

**Pull request merged**